### ISSUE TITLE
Installation issue: python 3.7.3 patch fails, all platforms

### ISSUE THREAD

**Issue by eugeneswalker** [LLNL/spack#17128]
`python@3.7.3` spec.yaml fails to build using `spack@develop` (commit id from 16-June-2020) on the following platforms:
* Ubuntu 18.04 amd64, ppc64le
* RHEL 7 amd64, ppc64le
* CentOS 7 amd64, ppc64le

Here is python-hyodd5.spec.yaml: [python-hyodd5.spec.yaml.txt](https://github.com/spack/spack/files/4790047/python-hyodd5.spec.yaml.txt)

```
$> spack install --file python-hyodd5.spec.yaml
...
2 out of 3 hunks FAILED -- saving rejects to file Lib/distutils/sysconfig.py.rej
==> Patch /opt/spack/var/spack/repos/builtin/packages/python/python-3.7+-distutils-C++.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/opt/spack/var/spack/repos/builtin/packages/python/python-3.7+-distutils-C++.patch' '-d' '.'
==> Error: Failed to install python due to ChildError: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/opt/spack/var/spack/repos/builtin/packages/python/python-3.7+-distutils-C++.patch' '-d' '.'Traceback (most recent call last):
  File "/opt/spack/lib/spack/spack/build_environment.py", line 819, in child_process
    return_value = function()
  File "/opt/spack/lib/spack/spack/installer.py", line 1063, in build_process
    pkg.do_patch()
  File "/opt/spack/lib/spack/spack/package.py", line 1222, in do_patch
    patch.apply(self.stage)
  File "/opt/spack/lib/spack/spack/patch.py", line 86, in apply
    apply_patch(stage, self.path, self.level, self.working_dir)
  File "/opt/spack/lib/spack/spack/patch.py", line 41, in apply_patch
    '-d', working_dir)
  File "/opt/spack/lib/spack/spack/util/executable.py", line 189, in __call__
    proc.returncode, long_msg)
spack.util.executable.ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/opt/spack/var/spack/repos/builtin/packages/python/python-3.7+-distutils-C++.patch' '-d' '.'
```

Also fails:
```
$> spack install python@3.7.3
...
<same error as above>
```

Looks like commit that broke this build is: 0095c4c25c459b5d7d25ce0462eed753b4036166
`python@3.7.3` builds fine at the commit right before it: 4b48f70393a20ef65f5e1394226d3829997431c5

@xjrc @adamjstewart @scottwittenburg

**Comment by scottwittenburg** [comment]
Ah shoot, I should have saved you the effort of digging into it by pinging you when I commented on #16856, which was the cause of the breakage.  Ugh.  Sorry. 🤦

**Comment by xjrc** [comment]
I'm working on this now and I should have a PR ready within the day (USA PST).

**Comment by adamjstewart** [comment]
Closed by #17140
